sunpy/coordinates/wcs_utils.py

```diff
@@ -34,7 +34,7 @@ def solar_wcs_frame_mapping(wcs):
     if hasattr(wcs, "coordinate_frame"):
         return wcs.coordinate_frame
 
-    dateobs = wcs.wcs.dateavg or wcs.wcs.dateobs or None
+    dateobs = wcs.wcs.dateavg or wcs.wcs.dateobs or wcs.wcs.datebeg or wcs.wcs.dateend or None
```

Comment:
Given we don't know where the WCS has come from, it makes sense to me to use the same hierarchy here as in `reference_date` in map.